go/ct/st/transient.go

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by the GNU Lesser General Public Licence v3
+//
+
+package st
+
+import (
+	. "github.com/Fantom-foundation/Tosca/go/ct/common"
+	"golang.org/x/exp/maps"
+)
+
+type Transient struct {
+	storage map[U256]U256
+}
+
+func (t *Transient) SetStorage(key U256, value U256) {
+	if t.storage == nil {
+		t.storage = make(map[U256]U256)
+		t.storage[key] = value
+	} else {
+		t.storage[key] = value
+	}
+}
+
+func (t *Transient) IsSet(key U256) bool {
+	_, ok := t.storage[key]
+	return ok
+}
+
+func (t *Transient) IsNotSet(key U256) bool {
```

Comment:
nit: i would delete this function; users should just call `IsSet` (or renamed to `IsEmpty`) and negate the result as needed.